Parse/src/main/java/com/parse/GCMService.java

```diff
@@ -116,19 +131,52 @@ private void handleGcmPushIntent(Intent intent) {
       String channel = intent.getStringExtra("channel");
 
       JSONObject data = null;
-      if (dataString != null) {
-        try {
-          data = new JSONObject(dataString);
-        } catch (JSONException e) {
-          PLog.e(TAG, "Ignoring push because of JSON exception while processing: " + dataString, e);
-          return;
+      try {
+        if (dataString != null) {
+            data = new JSONObject(dataString);
+        } else if (pushId == null && timestamp == null && dataString == null && channel == null) {
+          // The Parse SDK is older than GCM, so it has some non-standard payload fields.
+          // This allows the Parse SDK to handle push providers using the now-standard GCM payloads.
+          pushId = intent.getStringExtra("google.c.a.c_id");
+          String millisString = intent.getStringExtra("google.c.a.ts");
```

Comment:
Is this really the string google uses? We should extract these into constants (wouldn't be surprised if they are defined by GCM already)